sql/backfill.go

```diff
@@ -188,18 +193,44 @@ func (sc *SchemaChanger) getTableSpan() (roachpb.Span, error) {
 	}, nil
 }
 
+// backoffDurationDuringBackfill is the approximate time that the backfill
+// suspends activity between steps of the backfill. This value was picked
+// through some experimentation:
+// 1. Run block-writer against a 3 node cluster on GCE.
+// 2. Run create/drop index, create/drop column schema changes.
+var backoffDurationDuringBackfill = 75 * time.Millisecond
+
+// Slow down the backfill so that it doesn't rob all the processing power.
+// Do not backoff when backfillHasCompleted.
+func backoffDuringBackfill(backfillHasCompleted bool, testingKnobs *ExecutorTestingKnobs) {
+	if !backfillHasCompleted && !testingKnobs.DisableSchemaChangeBackfillBackoff {
+		delay := time.Duration(float64(backoffDurationDuringBackfill) * (0.75 + 0.5*rand.Float64()))
```

Comment:
Why is this randomized? When you're backing off due to contention on a shared resource, you want randomization so that that contention doesn't happen on the next iteration. Here it's more of a throttle than an exponential backoff so a fixed time seems reasonable.